plugins/crate-copy-azure/src/main/java/io/crate/copy/azure/AzureURI.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.copy.azure;
+
+import java.net.URI;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents URI provided by a user.
+ * @param account must be between 3 and 24 characters in length and may contain only numbers and lowercase letters.
+ * @param container must be between 3 and 63 characters in length  and may contain only numbers, lowercase letters and the hyphen character.
+ * @param endpoint which includes account name (for example, myaccount.blob.core.windows.net) without protocol (protocol is extracted from the setting).
+ * @param resourcePath path to the file
+ */
+public record AzureURI(
+    String account,
+    String container,
+    String endpoint,
+    String resourcePath
+) {
+
+    //                                                                        |          endpoint        |
+    //                                                                        |   account     | . |suffix|   container     |path
+    private static final Pattern URI_PATTERN = Pattern.compile("^(az://)(([a-z0-9]{3,24})\\.[^/]+)/([a-z0-9-]{3,63})(/.*)");
+
+
+    public static AzureURI of(URI uri) {
+        Matcher matcher = URI_PATTERN.matcher(uri.toString());
```

Comment:
Couldn't this make use of the already parsed URI properties? scheme, path, query, etc..

Reply:
https://github.com/crate/crate/pull/16631#discussion_r1763110554